packages/api/src/routes/medical/internal-cq.ts

```diff
@@ -10,35 +11,35 @@ import { Config } from "../../shared/config";
 import { asyncHandler } from "../util";
 
 const apiKey = Config.getCQApiKey();
-const apiMode = Config.getEnvType();
 
 dayjs.extend(duration);
 
 const router = Router();
 
 /**
- * GET /internal/carequality
+ * POST /internal/carequality/rebuild-directory
  *
  * Retrieves organizations from the Carequality Directory and uploads them into our database.
  * @returns Returns the number of organizations fetched, how many are newly-added and how many updated.
  */
-router.get(
-  "/",
+router.post(
+  "/rebuild-directory",
   asyncHandler(async (req: Request, res: Response) => {
-    const cq = new Carequality(apiKey, apiMode);
+    const cq = new Carequality(apiKey);
     const resp = await cq.listAllOrganizations();
-    const orgs = parseCQDirectoryEntries(resp.organizations);
+    const orgs = parseCQDirectoryEntries(resp);
 
     const response = {
-      totalFetched: resp.count,
+      totalFetched: resp.length,
       added: 0,
       updated: 0,
     };
 
-    for (const org of orgs) {
+    const directoryEntryPromises = executeAsynchronously(orgs, async org => {
       const dbResponse = await createOrUpdateCQDirectoryEntry(org);
       dbResponse.updated ? response.updated++ : response.added++;
-    }
+    });
+    await Promise.all([directoryEntryPromises]);
```

Comment:
This is not how we use `executeAsynchronously`, it doesn't return a list of promises, see other examples.

Reply:
I was referring to this example 

https://github.com/metriport/metriport/blob/develop/packages/api/src/external/commonwell/cq-bridge/coverage-enhancement-init.ts#L48-L57

Reply:
That example has 2 promises, one from `executeAsynchronously` and another from `getOrganizationOrFail`.

Check the return type of `executeAsynchronously`, it doesn't return an array - which is what `Promise.all` uses. So, if no array of promises, no need for `Promise.all`, right? Can check other examples, they are simpler to refer to.

Reply:
Yea, I overlooked that part! It's funny how there's so many other examples of how that function is used, and I happened to look at the one that was special 🪦